test/parallel/test-tls-psk-server.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+
+const tls = require('tls');
+const spawn = require('child_process').spawn;
+
+const CIPHERS = 'PSK+HIGH';
+const KEY = 'd731ef57be09e5204f0b205b60627028';
+const IDENTITY = 'TestUser';
+
+const server = tls.createServer({
+  ciphers: CIPHERS,
+  pskIdentity: IDENTITY,
+  pskCallback(identity) {
+    if (identity === IDENTITY) {
+      return {
+        psk: KEY
+      };
+    }
+  }
+});
+
+server.on('connection', common.mustCall((socket) => {}, 1));
+
+server.on('secureConnection', (socket) => {
+  socket.write('hello\r\n');
+
+  socket.on('data', (data) => {
+    socket.write(data);
+  });
+});
+
+let gotHello = false;
+let sentWorld = false;
+let gotWorld = false;
+let opensslExitCode = -1;
+
+server.listen(common.PORT, () => {
```

Comment:
s/`common.PORT`/`0`

note that in general we've been moving away from using `common.PORT` in the parallel tests.